modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -171,7 +171,11 @@ const deleteCellContents = (editor: Editor, rng: Range, cell: SugarElement<HTMLT
     const additionalCleanupNodes = Optionals.is(Traverse.parent(lastBlock), cell) ? [] : Traverse.siblings(lastBlock);
     Arr.each(additionalCleanupNodes.concat(Traverse.children(cell)), (node) => {
       if (!Compare.eq(node, lastBlock) && !Compare.contains(node, lastBlock)) {
-        Remove.remove(node);
+        // Don't clear any nodes that still have content in their descendants
+        const hasNonEmptyDescendant = PredicateExists.descendant(node, Fun.not(Empty.isEmpty));
+        if (!hasNonEmptyDescendant) {
+          Remove.remove(node);
+        }
```

Comment:
```suggestion
        if (Empty.isEmpty(node)) {
          Remove.remove(node);
        }
```
This isn't necessary and can be replaced with this I believe, as `Empty.isEmpty` already walks down the descendants, so this is just exponentially checking if child nodes are empty when it only needs to be done once.

- [ ] Don't use `PredicateExists.descendant` as it's going to cause child nodes to be checked multiple times.

Reply:
Actually this check could be moved up to line 173, instead of using the above suggestion.

Reply:
Yep nice, will do it all in there